digitalocean/config/config.go

```diff
@@ -76,25 +74,28 @@ func (c *Config) Client() (*CombinedConfig, error) {
 
 	userAgent := fmt.Sprintf("Terraform/%s", c.TerraformVersion)
 	var client *http.Client
+	var godoOpts []godo.ClientOpt
+
+	client = oauth2.NewClient(context.Background(), tokenSrc)
+	client.Transport = logging.NewTransport("DigitalOcean", client.Transport)
 
 	if c.HTTPRetryMax > 0 {
-		retryableClient := retryablehttp.NewClient()
-		retryableClient.RetryMax = c.HTTPRetryMax
-		retryableClient.RetryWaitMin = time.Duration(c.HTTPRetryWaitMin * float64(time.Second))
-		retryableClient.RetryWaitMax = time.Duration(c.HTTPRetryWaitMax * float64(time.Second))
+		retryConfig := godo.RetryConfig{
+			RetryMax:     3,
+			RetryWaitMin: godo.PtrTo(c.HTTPRetryWaitMin),
+			RetryWaitMax: godo.PtrTo(c.HTTPRetryWaitMax),
+		}
+
+		godoOpts = []godo.ClientOpt{godo.WithRetryAndBackoffs(retryConfig)}
 
-		client = retryableClient.StandardClient()
 		client.Transport = &oauth2.Transport{
 			Base:   client.Transport,
 			Source: oauth2.ReuseTokenSource(nil, tokenSrc),
```

Comment:
I was a little surprised I needed to maintain this transport. I thought having set the tokenSrc in [line 79 ](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1016/files#diff-698d451af8f49ff452e38c9ce233bcdd7af4278bbb353fa01c3a1abedfb59a19R79) would suffice. Without maintaing the transport, a nil error would occur in the godo client [here](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1016/files#diff-698d451af8f49ff452e38c9ce233bcdd7af4278bbb353fa01c3a1abedfb59a19R80).


I believe the `source.Source `gets lost when the transport gets set here [line 89](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1016/files#diff-698d451af8f49ff452e38c9ce233bcdd7af4278bbb353fa01c3a1abedfb59a19R80). When this is removed, you do not need to maintain the transport in [line 91](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1016/files#diff-698d451af8f49ff452e38c9ce233bcdd7af4278bbb353fa01c3a1abedfb59a19L87)

Reply:
With the changes that we made in godo, we should be able to remove the `client.Transport` modification here.